app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -180,7 +178,14 @@ class BrowserWebViewClient @Inject constructor(
                     }
                     false
                 }
-
+                is SpecialUrlDetector.UrlType.ShouldLaunchDuckPlayerLink -> {
+                    if (isRedirect) {
+                        webView.loadUrl(url.toString())
+                        return true
```

Comment:
I'm not following this. Are we trying to load the same URL that it's being loaded? If yes, you can just return false and don't call loadUrl.

Reply:
This forces `shouldInterceptRequest` with a YT URL to be called, otherwise that method is never executed and therefore Duck Player is never launched from a tracking URL

Reply:
I'd add a comment explaining the reasoning since the WebView docs mention to not call loadUrl with true.